formats/qcow-efi.nix

```diff
@@ -0,0 +1,46 @@
+{ config, lib, pkgs, modulesPath, ... }:
+{
+  # for virtio kernel drivers
+  imports = [
+    "${toString modulesPath}/profiles/qemu-guest.nix"
+  ];
+
+  options = {
+    diskSize = lib.mkOption {
+      default = "auto";
+      description = "The disk size in megabytes of the system disk image.";
+      type = with lib.types; oneOf [ ints.positive (enum [ "auto" ])];
+    };
+  };
+
+  config = {
+    fileSystems."/" = {
+      device = "/dev/disk/by-label/nixos";
+      autoResize = true;
+      fsType = "ext4";
+    };
+
+    fileSystems."/boot" = {
+      device = "/dev/disk/by-label/ESP";
```

Comment:
I am wondering if this should just be a small efi ESP partition just for the grub efi executable without kernel and mounted to `/boot/efi` with `boot.loader.grub.efiSysMountPoint = "/boot/efi";`. The advantage is that the kernel than can reside on the ext4 partition which might have more space and in general is a better/safer filesystem.

Reply:
cc @Lassulus